src/main/drivers/system.c

```diff
@@ -146,14 +146,24 @@ uint32_t micros(void)
     return (ms * 1000) + (usTicks * 1000 - cycle_cnt) / usTicks;
 }
 
-inline uint32_t getCycleCounter(void)
+inline int32_t getCycleCounter(void)
 {
     return DWT->CYCCNT;
 }
 
-uint32_t clockCyclesToMicros(uint32_t clockCycles)
+inline int32_t clockCyclesToMicros(int32_t clockCycles)
 {
-    return clockCycles / usTicks;
+    return clockCycles / (int32_t)usTicks;
+}
+
+inline int32_t clockCyclesTo10thMicros(int32_t clockCycles)
+{
+    return 10 * clockCycles / (int32_t)usTicks;
```

Comment:
Would it make sense to change `usTicks` into an `int32` instead of casting it most of the time?